kolibri/content/api.py

```diff
@@ -111,30 +122,48 @@ def intersection(queries):
         def union(queries):
             return reduce(lambda x, y: x | y, queries)
 
+        # all words with punctuation removed
+        all_words = [w for w in re.split('[?.,!";: ]', value) if w]
+        # words in all_words that are not stopwords
+        critical_words = [w for w in all_words if w not in stopwords_set]
```

Comment:
In some cases, `critical_words` or even `all_words` might be empty lists. Need to make sure these cases are handled (currently errors out).